src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+
+        }
+        Assert.assertFalse(inputItr.hasNext());
+        Assert.assertFalse(expectedItr.hasNext());
+    }
+
+    private boolean shouldFilterLine(VCFHeaderLine line, List<String> linesToIgnore) {
+        return (line != null)
+                && ((line instanceof VCFIDHeaderLine && linesToIgnore.contains(((VCFIDHeaderLine)line).getID()))
+                    || linesToIgnore.contains(line.getKey()));
+    }
+
+    private void runVariantAnnotatorAndAssertSomething(File input, File expected, List<String> additionalArguments, BiConsumer<VariantContext, VariantContext> assertion, String reference) throws IOException {
+        final File output = createTempFile("variantAnnotator", ".vcf");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        if (reference!=null) {
+            args.addReference(new File(reference));
+        }
+        args.addOutput(output).addArgument("V", input.getAbsolutePath());
+
+
+        // Handling a difference in syntax between GATK3 and GATK4 wrt. annotation groups
+        additionalArguments = additionalArguments.stream().map(a -> a.contains("Standard") ? a + "Annotation" : a).collect(Collectors.toList());
+        additionalArguments.forEach(args::add);
+
+        Utils.resetRandomGenerator();
+        runCommandLine(args);
+
+        assertHeadersMatch(output, expected, HEADER_LINES_TO_IGNORE);
+        final List<VariantContext> expectedVC = getVariantContexts(expected);
+        final List<VariantContext> actualVC = getVariantContexts(output);
+        assertForEachElementInLists(actualVC, expectedVC, assertion);
+    }
+
+    /**
+     * Returns a list of VariantContext records from a VCF file
+     *
+     * @param vcfFile VCF file
+     * @return list of VariantContext records
+     * @throws IOException if the file does not exist or can not be opened
+     */
+    private static List<VariantContext> getVariantContexts(final File vcfFile) throws IOException {
+        final VCFCodec codec = new VCFCodec();
+        final FileInputStream s = new FileInputStream(vcfFile);
+        final LineIterator lineIteratorVCF = codec.makeSourceFromStream(new PositionalBufferedStream(s));
+        codec.readHeader(lineIteratorVCF);
+
+        final List<VariantContext> VCs = new ArrayList<>();
+        while (lineIteratorVCF.hasNext()) {
+            final String line = lineIteratorVCF.next();
+            Assert.assertFalse(line == null);
+            VCs.add(codec.decode(line));
+        }
+
+        return VCs;
+    }
+
+    private static VCFHeader getHeaderFromFile(final File vcfFile) throws IOException {
+        SeekablePathStream stream = new SeekablePathStream(vcfFile.toPath());
+        VCFHeader header = VCFHeaderReader.readHeaderFrom(new SeekablePathStream(vcfFile.toPath()));
+        stream.close();
+        return header;
+    }
+
+    private static <T> void assertForEachElementInLists(final List<T> actual, final List<T> expected, final BiConsumer<T, T> assertion) {
```

Comment:
this shows up in 3 places in the code now, is there somewhere common it could live?

Reply:
there is an issue to extract a common interface for this, its just nontrivial enough that I haven't done it yet in any of the branches. So there it languishes....